hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/scale/AbstractSTestS3AHugeFiles.java

```diff
@@ -192,13 +201,7 @@ public void test_010_CreateHugeFile() throws IOException {
         true,
         uploadBlockSize,
         progress)) {
-      try {
-        streamStatistics = getOutputStreamStatistics(out);
-      } catch (ClassCastException e) {
-        LOG.info("Wrapped output stream is not block stream: {}",
-            out.getWrappedStream());
-        streamStatistics = null;
-      }
+      streamStatistics = getOutputStreamStatistics(out);
```

Comment:
why do we drop this? it wasn't right in the first place, so we should fail if we hit it?

Reply:
we will fail with an NPE now...in fact we already did further down. I don't remember why this catch/recovery is in place. I will add a requireNonNull for meaningful text if a failure ever surfaced though.